substrate/frame/assets/src/functions.rs

```diff
@@ -193,22 +193,31 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 			return Frozen
 		}
 		if let Some(rest) = account.balance.checked_sub(&amount) {
-			if let Some(frozen) = T::Freezer::frozen_balance(id.clone(), who) {
-				match frozen.checked_add(&details.min_balance) {
-					Some(required) if rest < required => return Frozen,
-					None => return Overflow,
-					_ => {},
-				}
-			}
+			match (
+				T::Holder::held_balance(id.clone(), who),
+				T::Freezer::frozen_balance(id.clone(), who),
+			) {
+				(None, None) =>
+					if rest < details.min_balance {
+						if keep_alive {
+							WouldDie
+						} else {
+							ReducedToZero(rest)
+						}
+					} else {
+						Success
+					},
+				(maybe_held, maybe_frozen) => {
+					let frozen = maybe_frozen.unwrap_or_default();
+					let held = maybe_held.unwrap_or_default();
 
-			if rest < details.min_balance {
-				if keep_alive {
-					WouldDie
-				} else {
-					ReducedToZero(rest)
-				}
-			} else {
-				Success
+					let required = frozen.saturating_sub(held).max(details.min_balance);
+					if rest < required {
+						return Frozen
```

Comment:
This `Frozen` is a `WithdrawConsequence` tied to the traits related to `tokens` (see [`WithdrawConsequence::Frozen`](https://docs.rs/frame-support/latest/frame_support/traits/tokens/enum.WithdrawConsequence.html#variant.Frozen) for more info). Changing its name would imply also modifying `pallet-assets-freezer`, `pallet-assets`, `pallet-balances` and (possibly, not sure) `pallet-broker`.

Additionally, the definition of what `Frozen` is, can be found in the terminology within [`tokens::fungible`](https://docs.rs/frame-support/latest/frame_support/traits/tokens/fungible/index.html#terminology) documentation:

> * **Frozen Balance:** A freeze on a specified amount of an account’s balance. Tokens that are frozen cannot be transferred.
> 
>   Multiple freezes always operate over the same funds, so they “overlay” rather than “stack”. This means that if an account has 3 freezes for 100 units, the account can spend its funds for any reason down to 100 units, at which point the freezes will start to come into play.